go/master/client.go

```diff
@@ -135,3 +135,77 @@ func (c *Client) taskFinished(taskID int) error {
 func (c *Client) NextRecord() []byte {
 	return <-c.ch
 }
+
+// NewEtcdClient create a new master client by etcd
+//
+// endpoints is the endpoints for etcd and separated by ",", such as
+// "172.0.1.0:2379,172.0.1.1:2379"
+// timeout is the timeout for etcd calls
+// bufSize is the record buffer size. NextRecord will read from this buffer.
+func NewEtcdClient(endpoints string, timeout int, bufSize int) *Client {
+	t := time.Second * time.Duration(timeout)
+	ep := strings.Split(endpoints, ",")
+	cli, err := clientv3.New(clientv3.Config{
+		Endpoints:   ep,
+		DialTimeout: t,
+	})
+	if err != nil {
+		log.Errorf("Init etcd connection failed: %v", err)
+		panic(err)
```

Comment:
Please return error here instead of panic, and don't need to log error. Caller will get the error and log it.